VirtualBlox/native.cpp

```diff
@@ -0,0 +1,79 @@
+#include "Adafruit_SSD1351.h"
+#include <unistd.h>
+#include <string.h>
+#include <SDL2/SDL.h>
+#include <iostream>
+
+extern void setup();
+extern void loop();
+
+FakeSerial Serial;
+FakeESP ESP;
+void* SPI;
+
+#define WIDTH 128
+#define HEIGHT 128
+ 
+using namespace std;
+SDL_Renderer *renderer;
+
+uint16_t pixels565[128*128];
+
+void Adafruit_SSD1351::drawRGBBitmap(int16_t x, int16_t y, uint16_t *bitmap,
+      int16_t w, int16_t h)
+{
+    memcpy(pixels565, bitmap, w * h * 2);
```

Comment:
this works because most of our demos use this to flush to the display and pass the reference to the entire canvas so we can access the image/buffer arg here. other demos don’t call this and so they would not work with this emulator right? 

not for this pr, but I wonder if it’s worth trying to abstract our graphics library even more so that each blockware implements this function under the hood. something like `BlockwareDisplay.flush()` which calls this under the hood or something? and it’d be required to include in the `loop()` of a demo to setup this integration? with the display and always calls flush at some point so we can hook into it here consistently?

Reply:
Yeah it's probably not complete, I just copied @mikeperiscope's approach. I think this is used to flush the `GFXCanvas` to the screen and so so anything that writes to the `GFXCanvas` should be handled by this method. But there are probably other screen drawing methods for partial refreshes that wouldn't be handled by this.